### PR TITLE
Show hint when label is floating

### LINKED ISSUE #53753: InputDecoration. Possibility to show hintText when unfocused and labelText is set
## Steps to Reproduce

```
import 'package:flutter/material.dart';

void main() => runApp(App());

class App extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Example(),
    );
  }
}

class Example extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: TextFormField(
          decoration: InputDecoration(
            floatingLabelBehavior: FloatingLabelBehavior.always,
            labelText: 'This is the labelText',
            hintText: 'Hello, World!',
          ),
        ),
      ),
    );
  }
}
```

**Expected results:**
* If the FloatingLabelBehavior is .always the hintText should not disappear and look similar to this:
IMG_0658

**Actual results:**
* hintText disappears, leaving a blank space.
IMG_0660

<details>
  <summary>Logs</summary>

```
[✓] Flutter (Channel dev, v1.16.4-pre.33, on Mac OS X 10.15.4 19E266, locale en-DE)
    • Flutter version 1.16.4-pre.33 at /Users/cedricgrothues/flutter
    • Framework revision aee9e94c21 (2 days ago), 2020-03-30 16:12:40 -0700
    • Engine revision c779894d08
    • Dart version 2.8.0 (build 2.8.0-dev.17.0 1402e8e1a4)

[!] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/cedricgrothues/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    ✗ Android SDK file not found: /Users/cedricgrothues/Library/Android/sdk/platforms/android-29/android.jar.
    • Try re-installing or updating your Android SDK,
      visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions.

 
[✓] Xcode - develop for iOS and macOS (Xcode 11.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4, Build version 11E146
    • CocoaPods version 1.8.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).

[✓] VS Code (version 1.43.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.8.1

 
[✓] Connected device (4 available)            
    • Cedric’s iPhone • 00008030-001218E421F0802E • ios            • iOS 13.4
    • macOS           • macOS                     • darwin-x64     • Mac OS X 10.15.4 19E266
    • Chrome          • chrome                    • web-javascript • Google Chrome 80.0.3987.149
    • Web Server      • web-server                • web-javascript • Flutter Tools

! Doctor found issues in 2 categories.

```

</details>

### LINKED ISSUE #60292: Hint text is not visible with FloatingLabelBehavior.always
@justinmc 
As mentioned in #30664 I'm expecting hint text to be visible with FloatingLabelBehavior.always and hintText provided

### PR DESCRIPTION
**Pull request by justinmc**
## Description

Previously, hint wasn't showing when floatingLabelBehavior was `always`, even though it had room. It only showed when focused:

Screen Shot 2020-06-26 at 3 40 00 PM
Screen Shot 2020-06-26 at 3 40 05 PM

This PR makes it show even when not focused.

## Related Issues

Closes https://github.com/flutter/flutter/issues/60292
Closes https://github.com/flutter/flutter/issues/53753

## Tests

I added the following tests:

Render an InputDecorator with `floatingLabelBehavior: FloatingLabelBehavior.always` and expect the hint to be visible even when it is not focused.

### CODE REVIEW

File: packages/flutter/lib/src/material/input_decorator.dart [hunk 1]
```diff
@@ -2086,9 +2086,12 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
   }
 
   // True if the label will be shown and the hint will not.
-  // If we're not focused, there's no value, and labelText was provided,
-  // then the label appears where the hint would.
-  bool get _hasInlineLabel => !widget._labelShouldWithdraw && decoration.labelText != null;
+  // If we're not focused, there's no value, labelText was provided, and 
+  // floatingLabelBehavior isn't set to always, then the label appears where the
+  // hint would.
+  bool get _hasInlineLabel => !widget._labelShouldWithdraw
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/material/input_decorator.dart:2092]: nit: the line is getting a bit too long. Maybe convert this to use curly brackets?

**Review by LongCatIsLooong — APPROVED** [review]
LGTM. I wasn't able to find anything on material.io that talks about this specific behavior though?

**Comment by flutter-github-sync** [comment]
Started Google testing for this PR

**Comment by flutter-github-sync** [comment]
Started Google testing for this PR

**Comment by flutter-github-sync** [comment]
Google testing passed!

**Comment by flutter-github-sync** [comment]
Google testing passed!

**Comment by flutter-github-sync** [comment]
Started Google testing for this PR

**Comment by flutter-github-sync** [comment]
Google testing passed!

**Comment by justinmc** [comment]
@LongCatIsLooong I couldn't find any specific guidelines about this behavior either.  It seems weird to me that the hint would suddenly appear on focus, but I wasn't able to prove that that's what the Material spec wants either.  I'll try to ask someone from Material.

**Comment by flutter-github-sync** [comment]
Started Google testing for this PR

**Comment by flutter-github-sync** [comment]
Google testing passed!

**Comment by justinmc** [comment]
I'm going to move forward with this since I haven't been able to get any definitive answer from Material. My best judgement says it should work like it does in this PR.  I'll be prepared to change this back if I hear otherwise.

**Comment by kjgasser** [comment]
So excited to see this :)

**Comment by diego-lipinski-de-castro** [comment]
is this on stable channel yet?

**Comment by justinmc** [comment]
Yes, as of the 1.20.0 release that just came out today!

**Pull request merged**